File: typescript/brands.ts
// Create the implementor class

export default interface Brand{
    boot():void;
  }
  
  // Concrete implementors
  
  // 1. Dell
  
  export class Dell implements Brand{
      boot(): void {
          console.log("Dell Computer booting ...")
      }
  }
  
  // 2. Hp
  
  export class HP implements Brand{
      boot(): void {
          console.log("Hp Computer booting ...")
      }
  }
  
  //3. Lenovo
  
  export class Lenovo implements Brand{
      boot(): void {
          console.log("Lenovo Computer booting ...")
      }
  }
  
  
  export class IBM implements Brand{
      boot(): void {
          console.log("IBM Computer booting ...")
      }
  }
  
  
  

File: python/brands.py
from abc import ABC, abstractmethod

# Implementor interface
class Brand(ABC):
    @abstractmethod
    def boot(self) -> None:
        pass

  

# Concrete Implementors
class Dell(Brand):
    def boot(self):
        print('Dell computer booting')

    
class Hp(Brand):
    def boot(self):
        print('Hp computer booting')


class Lenovo(Brand):
    def boot(self):
        print('Lenovo computer booting')
